doc/tools/coqrst/repl/ansicolors.py

```diff
@@ -91,7 +91,10 @@ def parse_ansi(code):
                  leading ‘^[[’ or the final ‘m’
     """
     classes = []
-    parse_style([int(c) for c in code.split(';')], 0, classes)
+    if code == "37":
+        pass # ignore white fg
+    else:
+        parse_style([int(c) for c in code.split(';')], 0, classes)
```

Comment:
I think it would be better to remove "white" from the list of results (or even remove the CSS class for it and leave this code untouched) and leave smarts about what number represents "white" to the "parse_color" function:

```
return ["ansi-" + cls for cls in classes if lcs != "white"]
```

Reply:
(or just undefine ansi-white)

Reply:
Undefining ansi-fg-white doesn't work because the default foreground color is #2e3436 (basically black).

Coqtop tags some things in white, as shown below ("<>" between "0" and "bottom" (with the check for "37" disabled), which we don't want.  Excluding the code string "37" prevents this while still allowing white foreground work for the errors case, which has the code string "37;41;1" for fg-white, bg-red and fg-bold.

![image](https://user-images.githubusercontent.com/1253341/110195977-24eef900-7df6-11eb-8393-bec7465ef66f.png)

The change I proposed does what's needed.  I didn't want to get into redefining how coqtop tags colors.  I suggest we apply the change as it is.  Or I'll withdraw the PR.  Up to you guys.

Reply:
OK, sounds good; please go ahead with the change.

> I didn't want to get into redefining how coqtop tags colors

But surely that's where the bug is, so maybe we should create an issue?

> Or I'll withdraw the PR. Up to you guys.

I think there are better ways to express frustration with the review process than "merge my PR or I'll withdraw it".

Reply:
Ah, but I'm not at all frustrated about this.  I will still sleep just fine if we don't fix this now.  If you want to suggest a fairly simple alternative of course I'll consider it.  But not everything worth doing is worth doing well.  My priorities are elsewhere.

(IMHO there's so much frustration and other negativity now due to the pandemic that it colors unrelated behavior and perceptions.)

The coloring is a feature of coqtop, not a bug, though the value of the feature is debatable given that coqtop is little used.  For example, this uses two shades of grey for different parts of the expression:

![image](https://user-images.githubusercontent.com/1253341/110219055-d3844f80-7e71-11eb-904c-3c9e700136dc.png)

These are defined in `Topfmt.default_tag_map`.  We could drop much of the coloring unconditionally or, I suppose we could add another command-line option to coqtop to not apply color within printed expressions (to be used when called from Sphinx).